spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -277,6 +279,7 @@ private void commitOperation(SnapshotUpdate<?> operation, String description) {
 
       try {
         long start = System.currentTimeMillis();
+        operation.toBranch(branch);
```

Comment:
We'll need to make sure the tests for this are exercising branches.

Reply:
+1, I don't see tests that would cover this.